src/main/java/dev/aziz/movies/services/MovieService.java

```diff
@@ -73,19 +66,45 @@ public MovieDto updateById(Long id, UpdateMovieDto updateMovieDto) {
             }
         }
 
-        if ("".equals(updateMovieDto.getGenre())) {
-            throw new AppException("Genre must not be empty.", HttpStatus.BAD_REQUEST);
-        } else if (updateMovieDto.getGenre() != null) {
-            movieById.setGenre(updateMovieDto.getGenre());
+        if (updateMovieDto.getGenres() != null) {
+            List<Genre> genreList = new ArrayList<>();
+            for (int i = 0; i < updateMovieDto.getGenres().size(); i++) {
```

Comment:
Instead of a for loop, can you use a for-each loop? Or even better with a Lambda?